pages/blog/_posts/banger.mdx

```diff
@@ -0,0 +1,279 @@
+---
+heading: A Deep Dive into a Video Rendering Pipeline
+showSubtitle: true
+subtitle: banger.show is a video app maker that heavily relies on background data processing
+image: /assets/blog/banger/banger-main.jpg
+date: 2024-05-01
+author: Igor Samokhovets
+disableCTA: true
+---
+
+import YouTube from 'react-youtube-embed';
+
+Hi everyone! My name is [Igor Samokhovets](https://twitter.com/IgorSamokhovets) and I'm a music producer who goes by the artist name Tequila Funk. Together with Mark Beziaev we created [banger.show](https://banger.show/) -- a video maker app for musicians, DJs, and labels. It allows music industry people to create stunning visual assets for their music.
+
+<YouTube id="eSLKm6IfZM4" height={300}/>
+
+ Creating a video for your new song takes only a few minutes and you don't need to know any complex software because banger.show works in your browser.
+
+
+## Making background processing snappy
+
+At banger.show, we do a lot of background processing. Managing render states, generating visual assets in the background, and even handling the rendering process on remote distributed workers. We chose Inngest because there's no better way to handle background jobs if you're using Next.js.
```

Comment:
> there's no better way to handle background jobs if you're using Next.js

This is fun, but I was wondering if there are specifics we could include, something like: "because it makes it easy to orchestrate multi-step workflows right in your Next.js codebase."

Reply:
Hi Dan! This would be more descriptive and correct:

> We chose Inngest because there's no better way to handle background jobs if you're using Next.js **without a custom server**. **It's a primary "flow controller" for us**, even though we have a simple queue solution based on Redis to handle tasks on our infra. It allows us to orchestrate, observe, and abstract away lower-level processes.

In simple terms Inngest allows us to operate on all of this low-level Redis stuff on a higher level, e.g. 

- not delegate every post-render task to the video rendering machine, but have some flow where the main server can receive render results from the workers and do something with it
- observe each step and its return data in the dashboard
- handle emergency cases when all of our infra goes down and we need to spin up some backup workers on AWS

Reply:
This is great context, thank you, Igor! I'll add these examples as well!